http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoMetricsMonitor.scala

```diff
@@ -0,0 +1,342 @@
+      r <- allRuntimes
+      c <- r.containers if c != RuntimeContainerServiceType.WelderService
+    } yield Map(
+      RuntimeStatusMetric(r.cloudContext.cloudProvider,
+                          c.imageType,
+                          r.status,
+                          getRuntimeUI(r.labels),
+                          getAzureCloudContext(r.cloudContext)
+      ) -> 1
+    )
+
+    allContainers.combineAll
+  }
+
+  /** Performs health checks for Running apps, and transforms to AppHealthMetric. */
+  private[monitor] def countAppsByHealth(
+    allClusters: List[KubernetesCluster]
+  )(implicit ev: Ask[F, AppContext]): F[Map[AppHealthMetric, Int]] = {
+    val allServices = for {
+      c <- allClusters if c.asyncFields.isDefined
+      n <- c.nodepools
+      // Only care about Running apps for health check metrics
+      a <- n.apps if a.status == AppStatus.Running
+      s <- a.appResources.services
+    } yield (c.cloudContext, c.asyncFields.get.loadBalancerIp, a, s.config.name)
+
+    allServices
+      .parTraverseN(parallelism) { case (cloudContext, baseUri, app, serviceName) =>
+        for {
+          ctx <- ev.ask
+          // For GCP just test if the app is available through the Leo proxy.
+          // For Azure impersonate the user and call the app's status endpoint via Azure Relay.
+          isUp <- cloudContext match {
+            case CloudContext.Gcp(project) => appDAO.isProxyAvailable(project, app.appName, serviceName)
+            case CloudContext.Azure(_) =>
+              for {
+                tokenOpt <- samDAO.getCachedArbitraryPetAccessToken(app.auditInfo.creator)
+                token <- F.fromOption(
+                  tokenOpt,
+                  AppCreationException(s"Pet not found for user ${app.auditInfo.creator}", Some(ctx.traceId))
+                )
+                authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, token))
+                relayPath = Uri.unsafeFromString(baseUri.asString) / app.appName.value
+                isUp <- serviceName match {
+                  case ServiceName("wds")      => wdsDAO.getStatus(relayPath, authHeader).handleError(_ => false)
+                  case ServiceName("cbas")     => cbasDAO.getStatus(relayPath, authHeader).handleError(_ => false)
+                  case ServiceName("cbas-ui")  => cbasUiDAO.getStatus(relayPath, authHeader).handleError(_ => false)
+                  case ServiceName("cromwell") => cromwellDAO.getStatus(relayPath, authHeader).handleError(_ => false)
+                  case s =>
+                    logger.warn(ctx.loggingCtx)(
+                      s"Unexpected app service encountered during health checks: ${s.value}"
+                    ) >> F.pure(false)
+                }
+              } yield isUp
+          }
+          // In addition to collecting aggregate metrics, log a warning for any app that is down.
+          _ <-
+            if (isUp) F.unit
+            else
+              logger.error(ctx.loggingCtx)(
```

Comment:
Right now this is just writing a log line -- my idea was we could hook this up to errorReporting or sentry in the future if we want to alert on apps that are down.

Reply:
That would be pretty awesome!
Counting apps that are down could be useful to display on grafana as well though

Reply:
Yup, I think both are complementary